File: tests/fixtures/code-sample.ts
// Sample TypeScript file for the code → diagram generator.
// Try:
//   diagram generate-class tests/fixtures/code-sample.ts --output class.mmd
//   diagram generate-tree  tests/fixtures/code-sample.ts --output tree.mmd
//   diagram generate-call  tests/fixtures/code-sample.ts --output call.mmd

export interface Shape {
    area(): number;
}

export class Circle implements Shape {
    constructor(public radius: number) {}
    area(): number {
        return Math.PI * this.radius * this.radius;
    }
}

export class Point {
    public x: number;
    public y: number;
}

export enum Color {
    Red,
    Green,
    Blue,
}

export function compute(p: Point): number {
    const a = p.area();
    return adjust(a, 1.0);
}

function adjust(v: number, by: number): number {
    return v + by;
}

import { HashMap } from "std-collections";

File: tests/fixtures/code-sample.rs
//! Sample Rust file for the code → diagram generator.
//!
//! Try:
//!   diagram generate-class examples/code-sample.rs --output sample-class.mmd
//!   diagram generate-tree  examples/code-sample.rs --output sample-tree.mmd
//!   diagram generate-call  examples/code-sample.rs --output sample-call.mmd

use std::collections::HashMap;

pub struct Point {
    pub x: f64,
    pub y: f64,
}

pub enum Color {
    Red,
    Green,
    Blue,
}

pub trait Shape {
    fn area(&self) -> f64;
}

impl Shape for Point {
    fn area(&self) -> f64 {
        self.x * self.y
    }
}

pub fn compute(p: Point) -> f64 {
    let a = p.area();
    let next = adjust(a, 1.0);
    finalise(next)
}

fn adjust(v: f64, by: f64) -> f64 {
    v + by
}

fn finalise(v: f64) -> f64 {
    v
}